### PR TITLE
DOC: Update GroupBy docstrings with See Also requirements

### PR DESCRIPTION
**Pull request by ktseng4096**
This PR fixes docstring validation errors described in https://github.com/pandas-dev/pandas/issues/58063

### CODE REVIEW

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -199,6 +199,15 @@ class providing the base-class of operations.
...
     Computed {fname} of values within each group.
 
+See Also
+--------
+Series.min : Return the min of the group values.
+DataFrame.min : Return the min of the group values.
+Series.max : Return the max of the group values.
+DataFrame.max : Return the max of the group values.
+Series.sum : Return the sum of the group values.
+DataFrame.sum : Return the sum of the group values.
```
Comment (rhshadrach) [inline pandas/core/groupby/groupby.py:209]: Why "group values" if you're linking to e.g. `Series.min`?

Comment (rhshadrach) [reply pandas/core/groupby/groupby.py:209]: Maybe just `Series values` or `DataFrame values` as appropriate?

Comment (ktseng4096) [reply pandas/core/groupby/groupby.py]: Thanks for catching that, I referenced the wrong function, should have been the groupby versions of those functions.

Comment (ktseng4096) [reply pandas/core/groupby/groupby.py]: @rhshadrach let me know if this is appropriate or not.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @ktseng4096

**Pull request merged**